plugins/cilium-cni/chaining/generic-veth/generic-veth.go

```diff
@@ -96,7 +96,12 @@ func (f *GenericVethChainer) Add(ctx context.Context, pluginCtx chainingapi.Plug
 
 			addrsv6, err := netlink.AddrList(link, netlink.FAMILY_V6)
 			if err == nil && len(addrsv6) > 0 {
-				vethIPv6 = addrsv6[0].IPNet.IP.String()
+				for _, addrv6 := range addrsv6 {
+					if !addrv6.IP.IsLinkLocalUnicast() {
```

Comment:
I believe this should be `if addrv6.IP.IsGlobalUnicast() {`

Reply:
OK, the `if addrv6.IP.IsGlobalUnicast() {` should be better.